chia/daemon/keychain_proxy.py

```diff
@@ -89,13 +97,80 @@ def format_request(self, command: str, data: Dict[str, Any]) -> WsRpcMessage:
 
         return super().format_request(command, data)
 
+    async def _get(self, request: WsRpcMessage) -> WsRpcMessage:
+        """
+        Overrides DaemonProxy._get() to handle the connection state
+        """
+        try:
+            if not self.disconnect:  # if we are disconnected, and we send a request we should throw original error.
+                await asyncio.wait_for(self.connection_established.wait(), timeout=10)  # is 10 seconds too long?
+            else:
+                self.log.error("Attempting to send request to a keychain-proxy that has shut down.")
+            self.log.debug(f"Sending request to keychain command: {request['command']} from {request['origin']}.")
+            return await super()._get(request)
+        except asyncio.TimeoutError:
+            raise KeychainProxyConnectionFailure("Could not reconnect to Keychain!")
+
+    async def start(self) -> None:
+        self.keychain_connection_task = asyncio.create_task(self.connect_to_keychain())
+        await self.connection_established.wait()  # wait until connection is established.
+
+    async def connect_to_keychain(self) -> None:
```

Comment:
Can we add some logging in here to indicate that we're attempting to connect/reconnect?

Reply:
added